src/Moryx.Communication.Mqtt.Endpoints/ResourceRpcEndpoint.cs

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2025, Phoenix Contact GmbH & Co. KG
+// Licensed under the Apache License, Version 2.0
+
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using Moryx.AbstractionLayer.Identity;
+using Moryx.AbstractionLayer.Resources;
+using Moryx.Communication.Mqtt.Builders;
+using Moryx.Communication.Mqtt.Components;
+using Moryx.Communication.Mqtt.Converters;
+using Moryx.Serialization;
+using MQTTnet;
+
+namespace Moryx.Communication.Mqtt.Endpoints;
+
+/// <summary>
+/// Endpoint to remote call a procedure/method on a resource
+/// </summary>
+public class ResourceRpcEndpoint(
+    ILogger<ResourceRpcEndpoint> logger,
+    IResourceManagement resourceManagement,
+    MqttClientUserOptions options)
+    : IMqttEndpoint
+{
+    private const string message = "No resource or method found with the given Identifier:`{id}`, method:`{method}`. Task and Task<T> are not supported return types.";
```

Comment:
I would prefer having dedicated error(message)s. You have three different errors here
- Resource not found
- Method found, but has unsupported return type Task
- Method not found on resource

Combined with early returns that would further improve the map method